packages/nc-gui/components/erd/FullScreenToggle.vue

```diff
@@ -0,0 +1,28 @@
+<script lang="ts" setup>
+import { Panel, PanelPosition } from '@vue-flow/additional-components'
+import type { ERDConfig } from './utils'
+import MiFullscreen from '~icons/material-symbols/fullscreen'
+import MiFullscreenExit from '~icons/material-symbols/fullscreen-exit'
+
+const props = defineProps<{
+  config: ERDConfig
+}>()
+
+const config = useVModel(props, 'config')
+
+const toggleFullScreen = () => {
+  config.value.isFullScreen = !config.value.isFullScreen
+}
```

Comment:
Here it's mutating the component prop directly which is a bad practice. And `useVModel` is not necessary if 2-way binding is not used.

#### Suggestions

- You can use 2 way binding and do something as follows
    ```js
    config.value = { ...config.value, isFullScreen: !config.value.isFullScreen }
     ```
- Or you can emit an event and mutate the prop from the parent.

Reply:
@pranavxc  Taken the second approach